instrumentation/opentelemetry-instrumentation-aiohttp-client/setup.py

```diff
@@ -18,20 +18,42 @@
 
 
 import os
+from configparser import ConfigParser
 
 import setuptools
 
+config = ConfigParser()
+config.read("setup.cfg")
+
+# We provide extras_require parameter to setuptools.setup later which
+# overwrites the extra_require section from setup.cfg. To support extra_require
+# secion in setup.cfg, we load it here and merge it with the extra_require param.
+extras_require = {}
+if "options.extras_require" in config:
+    for key, value in config["options.extras_require"].items():
+        extras_require[key] = [v for v in value.split("\n") if v.strip()]
+
 BASE_DIR = os.path.dirname(__file__)
-VERSION_FILENAME = os.path.join(
+PACKAGE_FILENAME = os.path.join(
     BASE_DIR,
     "src",
     "opentelemetry",
     "instrumentation",
     "aiohttp_client",
-    "version.py",
+    "package.py",
 )
 PACKAGE_INFO = {}
-with open(VERSION_FILENAME) as f:
+with open(PACKAGE_FILENAME) as f:
     exec(f.read(), PACKAGE_INFO)
 
-setuptools.setup(version=PACKAGE_INFO["__version__"])
+# Mark any instruments/runtime dependencies as test dependencies as well.
+extras_require["instruments"] = PACKAGE_INFO["_instruments"]
```

Comment:
Might be a dumb question but why can't we put the `_instruments` directly in the `options.extras_require` inside `setup.cfg`?

Reply:
We can but this data is used in two places. In setup.cfg/py for pip to use and by instrumentor to do runtime checks. It is very similar to how we refer to `__version__` of each package. It is needed by setup.py and also by the python code in package itself. So we store it in python and read it from the Python code inside setup.py. 

We can still store it only in setup.cfg and then use a module like `pkg_resource` to read the package metadata and load this information but I think that'd be unnecessary complexity and possibly perf issues.